tests/RouteTest.php

```diff
@@ -300,6 +303,184 @@ public function testMiddlewareCallSuccessArrayWithoutContainerLL(): void
+                    $group->addRoute(Route::get('/post/{id}')->name('api-v2-news-post/view'));
+                }));
+                $group->addGroup(new Group('/blog', function (Group $group) {
+                    $group->addRoute(Route::get('/post')->name('api-v2-blog-post/index'));
+                    $group->addRoute(Route::get('/post/{id}')->name('api-v2-blog-post/view'));
+                }));
+                $group->addRoute(Route::get('/note')->name('api-v2-note/index'));
+                $group->addRoute(Route::get('/note/{id}')->name('api-v2-note/view'));
+            }));
+        }, $container);
+
+        $items = $apiGroup->getItems();
+
+        $func = function ($item) use (&$func) {
+            $this->assertSame(true, $item->hasContainer());
+            if ($item instanceof Group) {
+                $items = $item->getItems();
+                array_walk($items, $func);
+            }
+        };
+        array_walk($items, $func);
+    }
+
+    public function testContainerAutoInjectionByRouterFactory()
+    {
+        $container = $this->getContainer([
+            TestController::class => new TestController(),
+        ]);
+
+        $routes = [
+            Route::get('/info')->name('api-info'),
+            Group::create('/v2', [
+                Route::get('/user')->name('api-v2-user/index'),
+                Route::get('/user/{id}')->name('api-v2-user/view'),
+                Group::create('/news', [
+                    Route::get('/post')->name('api-v2-news-post/index'),
+                    Route::get('/post/{id}')->name('api-v2-news-post/view'),
+                ]),
+                Group::create('/blog', [
+                    Route::get('/post')->name('api-v2-blog-post/index'),
+                    Route::get('/post/{id}')->name('api-v2-blog-post/view'),
+                ]),
+                Route::get('/note')->name('api-v2-note/index'),
+                Route::get('/note/{id}')->name('api-v2-note/view'),
+            ]),
+            Group::create('/v2', [
+                Route::get('/user')->name('api-v2-user/index'),
+                Route::get('/user/{id}')->name('api-v2-user/view'),
+                Group::create('/news', [
+                    Route::get('/post')->name('api-v2-news-post/index'),
+                    Route::get('/post/{id}')->name('api-v2-news-post/view'),
+                ]),
+                Group::create('/blog', [
+                    Route::get('/post')->name('api-v2-blog-post/index'),
+                    Route::get('/post/{id}')->name('api-v2-blog-post/view'),
+                ]),
+                Route::get('/note')->name('api-v2-note/index'),
+                Route::get('/note/{id}')->name('api-v2-note/view'),
+            ])
+        ];
+
+        $factory = new RouterFactory($this->getEngineFactory(), $routes);
+        $router = $factory($container);
+        $items = $router->getItems();
+
+        $func = function ($item) use (&$func) {
```

Comment:
This method must be extracted as a private method, because it's used in at least 3 methods.

Reply:
Ok